src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2878,18 +2876,17 @@ static bool createEntity
       {
         attrType = defaultType(orion::ValueTypeVector);
       }
-
-      bsonAttr.append(ENT_ATTRS_TYPE, attrType);
     }
     else
     {
-      bsonAttr.append(ENT_ATTRS_TYPE, attrsV[ix]->type);
+      attrType = attrsV[ix]->type;
```

Comment:
Not sure how it would look ...
However, if you initialize `attrType` with `attrsV[ix]->type`, then you can remove the entire 'else' part.
Up to you

Reply:
I get you point... but I think I prefer to make explicit how the program flow ends in the case of else in this case.

NTC